lib/remap-functions/src/parse_json.rs

```diff
@@ -9,63 +9,41 @@ impl Function for ParseJson {
     }
 
     fn parameters(&self) -> &'static [Parameter] {
-        &[
-            Parameter {
-                keyword: "value",
-                accepts: |v| matches!(v, Value::Bytes(_)),
-                required: true,
-            },
-            Parameter {
-                keyword: "default",
-                accepts: |v| matches!(v, Value::Bytes(_)),
-                required: false,
-            },
-        ]
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Bytes(_)),
+            required: true,
+        }]
     }
 
     fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
         let value = arguments.required("value")?.boxed();
-        let default = arguments.optional("default").map(Expr::boxed);
 
-        Ok(Box::new(ParseJsonFn { value, default }))
+        Ok(Box::new(ParseJsonFn { value }))
     }
 }
 
 #[derive(Debug, Clone)]
 struct ParseJsonFn {
     value: Box<dyn Expression>,
-    default: Option<Box<dyn Expression>>,
 }
 
 impl Expression for ParseJsonFn {
     fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
-        let to_json = |value: Value| {
-            let bytes = value.unwrap_bytes();
-            let value = serde_json::from_slice::<'_, Value>(&bytes)
-                .map_err(|e| format!("unable to parse json: {}", e))?;
-
-            Ok(value)
-        };
-
-        crate::util::convert_value_or_default(
-            self.value.execute(state, object),
-            self.default.as_ref().map(|v| v.execute(state, object)),
-            to_json,
-        )
+        let value = self.value.execute(state, object)?;
+        let bytes = value.unwrap_bytes();
```

Comment:
```suggestion
        let bytes = value.try_bytes()?;
```

I know you didn't change this in this PR, but we should avoid unwrapping if we aren't 100% sure this will always be bytes.